src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -1303,24 +1330,23 @@ double VideoDecoder::getPtsSecondsForFrame(
   return ptsToSeconds(stream.allFrames[frameIndex].pts, stream.timeBase);
 }
 
-void VideoDecoder::convertFrameToBufferUsingSwsScale(
-    RawDecodedOutput& rawOutput) {
-  AVFrame* frame = rawOutput.frame.get();
-  int streamIndex = rawOutput.streamIndex;
+int VideoDecoder::convertFrameToBufferUsingSwsScale(
+    int streamIndex,
+    const AVFrame* frame,
+    torch::Tensor& outputTensor) {
```

Comment:
I have updated the signature of `convertFrameToBufferUsingSwsScale` for it to be very similar to `convertFrameToTensorUsingFilterGraph`. The signatures are now:

```
int VideoDecoder::convertFrameToBufferUsingSwsScale(
    int streamIndex,
    const AVFrame* frame,
    torch::Tensor& outputTensor);

torch::Tensor VideoDecoder::convertFrameToTensorUsingFilterGraph(
    int streamIndex,
    const AVFrame* frame);
```

Their signatures being so different previously was a constant source of confusion for me.

I realize that this change seemingly goes against our TODO of folding the pre-allocated tensor within `RawOutput`. But I think this TODO will in fact be easier to address with that change, as both APIs will be more closely aligned.

Reply:
Why is this function now returning the height?